### PR TITLE
Autogenerate fixture data

### PR DESCRIPTION
**Pull request by mloughran**
As suggested in [#45](https://github.com/jeromegn/protobuf.cr/pull/45) I had a look at autogenerating test fixtures. This definitely seems worthwhile given that the `.proto` files and supposedly generated `.pb.cr` have diverged considerably.

This PR also fixes autogeneration of empty messages which was required to avoid manually editing the generated files.

Regenerating `make fixtures` is currently a no-op. It would probably be sensible to add a github action to assert this to avoid regressions.

### CODE REVIEW

File: spec/fixtures/test3.pb.cr [hunk 1]
```diff
@@ -34,36 +34,13 @@ module TestMessagesV3
...
-      required :parent_process_pid, :int32, 7
-      required :parent_process_uuid, :string, 8
-      required :account_name, :string, 9
-      required :domain_name, :string, 10
-      optional :command_line, :string, 11
-      optional :site_id, :string, 12
-      optional :uuid, :string, 13
-      optional :process_writer, :string, 14
-    end
   end
```
Comment (mloughran) [inline spec/fixtures/test3.pb.cr:46]: This does not seem to be used – presumably legacy?

File: src/protobuf/generator.cr [hunk 2]
```diff
@@ -329,13 +329,13 @@ module Protobuf
...
 
+        puts "contract_of \"#{syntax}\" do"
         unless message_type.field.nil?
-          puts "contract_of \"#{syntax}\" do"
           indent do
             message_type.field.not_nil!.each { |f| field!(f, syntax) } unless message_type.field.nil?
           end
-          puts "end"
         end
+        puts "end"
```
Comment (jgaskins) [inline src/protobuf/generator.cr:338]: Good call on this. I didn't notice that the `contract_of` macro was what defines several of the necessary methods.

File: spec/fixtures/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+test.pb.cr:
+	protoc --crystal_out=. --plugin=protoc-gen-crystal=../../bin/protoc-gen-crystal test.proto
+	sed -i '' 's/^require/# &/g' test.pb.cr
+
+test3.pb.cr:
+	protoc --crystal_out=. --plugin=protoc-gen-crystal=../../bin/protoc-gen-crystal test3.proto
+	sed -i '' 's/^require/# &/g' test3.pb.cr
+
+clean:
+	rm -f $(outputs)
```
Comment (jgaskins) [inline spec/fixtures/Makefile:17]: This is fantastic!

**Review by jgaskins — APPROVED** [review]
This is great! Thank you!

**Pull request merged**